### PR TITLE
Add loan actions table documentation

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Add a table with the loan actions and requirements to the main `pallet-loan` docs. I think this can be helpful for a new reader of this pallet.

*I've already supposed the refactor done in #1155*

## Changes and Descriptions

- Added documentation table

## Type of change

- Documentation

# How Has This Been Tested?

```sh
cargo doc -p pallet-loans
```

# Checklist:

- [ ] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I rebased on the latest `main` branch

### CODE REVIEW

File: pallets/loans/src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,20 @@
...
+//! |      Action     |   From  |    To   |      Role     | Collateral Owner |
+//! |-----------------|---------|---------|---------------|------------------|
+//! |      create     |         | Created |    Borrower   |        Yes       |
+//! |      price      | Created |  Active | PrincingAdmin |                  |
+//! |      extend     |  Active |  Active | PrincingAdmin |                  |
+//! |      borrow     |  Active |  Active |               |        Yes       |
+//! |      repay      |  Active |  Active |               |        Yes       |
+//! |    write_off    |  Active |  Active |               |                  |
+//! | admin_write_off |  Active |  Active |   LoanAdmin   |                  |
+//! |      close      |  Active |  Close  |               |        Yes       |
```
Comment (offerijns) [inline pallets/loans/src/lib.rs:29]:
```suggestion
//! |      Action     |   From  |    To   |      Role     | Collateral Owner |
//! |-----------------|---------|---------|---------------|------------------|
//! |      create     |         | Created |    Borrower   |        Yes       |
//! |      price      | Created |  Active | PricingAdmin |                  |
//! |      borrow     |  Active |  Active |               |        Yes       |
//! |      repay      |  Active |  Active |               |        Yes       |
//! |    write_off    |  Active |  Active |               |                  |
//! | admin_write_off |  Active |  Active |   LoanAdmin   |                  |
//! |      close      |  Active |  Close  |               |        Yes       |
```

**Review by offerijns — COMMENTED** [review]
Great idea to add this!

File: pallets/loans/src/lib.rs (hunk 1 above)
Comment (offerijns) [reply pallets/loans/src/lib.rs:29]: Just removed `extend` (as it doesn't exist today) and fixed a few typos.

Comment (lemunozm) [reply pallets/loans/src/lib.rs:29]: Great, thanks!

File: pallets/loans/src/lib.rs [hunk 2]
```diff
@@ -14,6 +14,19 @@
...
+//!
+//! |      Action     |   From  |    To   |      Role     | Collateral Owner |
+//! |-----------------|---------|---------|---------------|------------------|
+//! |      create     |         | Created |    Borrower   |        Yes       |
+//! |      price      | Created |  Active | PricingAdmin |                  |
+//! |      borrow     |  Active |  Active |               |        Yes       |
+//! |      repay      |  Active |  Active |               |        Yes       |
+//! |    write_off    |  Active |  Active |               |                  |
+//! | admin_write_off |  Active |  Active |   LoanAdmin   |                  |
+//! |      close      |  Active |  Closed  |               |        Yes       |
```
Comment (offerijns) [inline pallets/loans/src/lib.rs:28]:
```suggestion
//! |      create     |         | Created |    Borrower   |        Yes       |
//! |      price      | Created |  Active | PricingAdmin  |                  |
//! |      borrow     |  Active |  Active |               |        Yes       |
//! |      repay      |  Active |  Active |               |        Yes       |
//! |    write_off    |  Active |  Active |               |                  |
//! | admin_write_off |  Active |  Active |   LoanAdmin   |                  |
//! |      close      |  Active |  Closed |               |        Yes       |
```

Comment (offerijns) [reply pallets/loans/src/lib.rs:28]: Looks like I broke the indentation

Comment (lemunozm) [reply pallets/loans/src/lib.rs:28]: I've seen it later, I'll fix it in any of the following PRs. No worries 👍🏻

**Pull request merged**